### PR TITLE
refs #1979:  add workaround for toolchain JS files that were referenced wrong

### PR DESCRIPTION
**Pull request by syphar**
See #1979

### CODE REVIEW

File: src/web/rustdoc.rs [hunk 1]
```diff
@@ -58,6 +58,37 @@ pub(crate) struct RustdocRedirectorParams {
     target: Option<String>,
 }
 
+/// try to serve a toolchain specific assert from the legacy location.
```
Comment (Nemo157) [inline src/web/rustdoc.rs:61]:
```suggestion
/// try to serve a toolchain specific asset from the legacy location.
```

Comment (syphar) [reply src/web/rustdoc.rs]: the ole muscle memory :D 

Fixed it

File: src/web/rustdoc.rs [hunk 2]
```diff
@@ -210,7 +213,12 @@ pub(crate) async fn rustdoc_redirector_handler(
                     {
                         debug!(?target, ?err, "got error serving file");
                     }
-                    return Err(AxumNope::ResourceNotFound);
+                    // FIXME: we sometimes still get requests for toolchain
+                    // specific static assets under the crate/version/ path.
+                    // This is fixed in rustdoc, but pending a rebuild for
+                    // docs that were affected by this bug.
+                    // https://github.com/rust-lang/docs.rs/issues/1979
+                    return try_serve_legacy_toolchain_asset(storage, config, target).await;
```
Comment (Nemo157) [inline src/web/rustdoc.rs:221]: To avoid letting another file regress we could check that the filename starts with `search-2021` or `search-2022`.

Comment (syphar) [reply src/web/rustdoc.rs]: I updated the code, checking for a `search-` prefix

**Review by Nemo157 — APPROVED** [review]

**Pull request merged**